### DISCUSSION TITLE
How to match different unique Docker tags within a single repository

### DISCUSSION THREAD

**Discussion by kjq** [renovatebot/renovate#20277] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

We have a single Docker repository (`repo`) that can hold many different images by a custom tag (`service`-`version`).

What we would like to do is match the versions based on the Dockerfile.

Before:
`FROM AWS_REPO/REPO_NAME:PROJECT-SERVICEA-1.0.0` 

and update this version with the corresponding version in the repository so it becomes:

After:
`FROM AWS_REPO/REPO_NAME:PROJECT-SERVICEA-1.0.1`

What is happening is it either fails to recognize the value or blindly updates all the containers with one container instead of matching the tag name.  This would be similar to if I had a tag 1.0.0-alpine and I only want to match that version `-alpine`.

Our repository looks like this:

```
REPO_NAME
  |---PROJECT-SERVICEA-1.0.0
  |---PROJECT-SERVICEA-1.0.1
  |---PROJECT-serviceB-1.0.0
```

The renovate output for one of those Docker containers looks like this:

```json
{
             "packageFile": "engine/Dockerfile",
             "deps": [
               {
                 "depName": "AWS REPO/REPO_NAME",
                 "currentValue": "PROJECT-SERVICEA-1.0.1",
                 "replaceString": "AWS REPO/REPO_NAME:PROJECT-SERVICEA-1.0.1",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "final",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [],
                 "versioning": "docker",
                 "skipReason": "invalid-value"
               }
             ]
           },
```

### What you tried so far.

We are using a package rule like this:

```yaml
{
            "groupName": "Custom Containers",
            "versioning": "regex:^(?<project>.*)-?(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)?$",
            "prPriority": 5,
            "extends": [
                "schedule:nonOfficeHours",
                "schedule:daily"
            ],
            "matchDatasources": [
                "docker"
            ],
            "matchPackagePatterns": [
                "(?<project>.*)-(?<version>.*)$"
            ]
        },
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
You could try using `loose` versioning
